RFC/src/RFC-0132_Merge_Mining_Monero.md

```diff
@@ -0,0 +1,172 @@
+
+The protocol must enable a Tari base node to make the following assertions:
+
+**REQ 1.** The achieved mining difficulty exceeds target difficulty as dictated by Tari consensus, 
+
+**REQ 2.** The Monero block was constructed before the Tari block. This is to prevent a miner from submitting blocks from the parent chain that satisfy the auxiliary chain's difficulty without doing new work.
+   
+It's worth noting that a Tari base node never has to contact or download data from Monero to make these assertions.
+
+## Merge Mining on Tari
+
+A new Tari block template is obtained from a Tari base node by calling the `get_new_block_template` GRPC method, setting `Monero` as the chosen PoW algorithm.
+The `Monero` algorithm must be selected so that the correct mining difficulty for the Monero algorithm is returned. Remember, that Monero and SHA difficulties 
+are independent (See [RFC-0131_Mining.html](/RFC-0131_Mining.html)).
+
+Next, construct a _Blake256_ hash of _some_ of the Tari header fields. We'll call this hash the merge mining hash \\( h_m \\) that commits to 
+the following header fields in order: `version`, `height`,`prev_hash`,`timestamp`,`output_mr`,`range_proof_mr`,`output_mmr_size`,`kernel_mr`,
+`kernel_mmr_size`,`total_kernel_offset`,`total_script_offset`. Note, this hash does not include the `pow` and `nonce` fields, as these fields are set as part of mining.
+
+To have the chance of mining a Monero block as well as a Tari block, we must obtain a new valid monero block template, by calling [get_block_template].
+This returns a `blocktemplate_blob`, that is, a serialized Monero block containing the Monero block header, coinbase and a list of hashes referencing the 
+transactions included in the block. Additionally, a `blockhashing_blob` is a fixed size blob containing `serialized_monero_header`, `merkle_tree_root` and 
+`txn_count` concatenated together. The `merkle_tree_root` is a _merkle root_ of the coinbase + the transaction hashes contained in the block.
+
+```rust,ignore
+pub struct Block {
+    /// The block header
+    pub header: BlockHeader,
+    /// Coinbase transaction a.k.a miner transaction
+    pub miner_tx: Transaction,
+    /// References to the transactions included in this block
+    pub tx_hashes: Vec<hash::Hash>,
+}
+```
+_fig 1. The Monero block struct_
+
+Next, modify the Monero block template by including the merge mining hash \\( h_m \\) in the extra fields of the coinbase transaction. Monero has a [merge mining subfield] 
+to accommodate this data. Importantly, the extra field data part of the coinbase transaction hash and therefore the `merkle_tree_root`, the `blockhashing_blob` must be 
+reconstructed. A rust port of [Monero's tree hash] algorithm is needed to achieve this. The coinbase hash MUST be the first element to be hashed when constructing the `merkle_tree_root`.
+This satisfies **REQ 2**, proving that the Tari block was constructed before the parent chain's block. 
```

Comment:
Is the order correct here? Seems to contradict REQ 2

Reply:
Maybe this is better

```suggestion
This satisfies **REQ 2**, proving that the proof-of-work was performed for the Tari block.
```

Reply:
Done